Library/Homebrew/formulary.rb

```diff
@@ -155,6 +157,34 @@ def self.class_s(name)
     class_name
   end
 
+  def self.formula_from_json(name, json)
+    hash = JSON.parse json
+    bottle_tag = MacOS.version.to_sym.to_s
+
+    unless hash.key? bottle_tag
+      raise FormulaUnreadableError.new(name, "No bottle informtion for this operating system.")
+    end
+
+    resource = Resource.new name do
+      url hash[bottle_tag]["url"]
+      sha256 hash[bottle_tag]["sha256"]
+    end
+
+    resource.fetch unless resource.downloaded?
+
+    formula_file = "#{name}/*/.brew/#{name}.rb"
+    contents = Utils.popen_read "tar", "--extract", "--to-stdout", "--file", resource.cached_download, formula_file
+
+    # The formula file in the bottle archive doesn't contain bottle information so we need to insert it manually
+    contents.sub!(/(^class .* < Formula$)/, "\\1\n#{<<~BOTTLE_BLOCK}")
+      bottle do
+        sha256 cellar: #{hash[bottle_tag]["cellar"]}, #{bottle_tag}: "#{hash[bottle_tag]["sha256"]}"
+      end
+    BOTTLE_BLOCK
```

Comment:
This feels a bit hacky... not sure if there's a better way to handle it.

Reply:
Out of interest: why does this formula need a bottle block at all? 

An alternative way to handle this: read the formula file and then do a `class_eval` after the fact to do the `bottle do` there.

Reply:
> Out of interest: why does this formula need a bottle block at all?

If there's no bottle block it installs from source.



> An alternative way to handle this: read the formula file and then do a `class_eval` after the fact to do the `bottle do` there.

Yeah, that's more graceful for sure. The reason I didn't do that here is so all of the JSON handling logic could occur in one place. Since this method only generates the formula contents as a string, it's not going to be evaluated for a bit more. The `class_eval` (which TBH isn't something I'm super familiar with) would need to happen after the new class is created.